test/sanity/run-test.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Copyright 2022 The Ceph CSI Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function cleanup {
+  echo 'pkill -f cephcsi'
+  if [ -z "$GITHUB_ACTIONS" ]
+  then
+    # if not running on github actions, do not use sudo
+    pkill -f cephcsi
+  else
+    # if running on github actions, use sudo
+    sudo pkill -f cephcsi
+  fi
+  echo 'Deleting CSI sanity test binary'
+  rm -rf csi-test
+}
+
+trap cleanup EXIT
+
+function install_csi_sanity_bin {
+  echo 'Installing CSI sanity test binary...'
+  mkdir -p $GOPATH/src/github.com/kubernetes-csi
+  pushd $GOPATH/src/github.com/kubernetes-csi
+  export GO111MODULE=off
+  git clone https://github.com/kubernetes-csi/csi-test.git -b v4.3.0
+  pushd csi-test/cmd/csi-sanity
+  make install
+  popd
+  popd
+}
+
+if [[ -z "$(command -v csi-sanity)" ]]; then
+	install_csi_sanity_bin
+fi
+
+readonly endpoint='unix:///csi/csi.sock'
+nodeid='node.com'
+rbddrivername="rook-ceph.rbd.csi.ceph.com"
+cephfsdrivername="rook-ceph.cephfs.csi.ceph.com"
+if [[ "$#" -gt 0 ]] && [[ -n "$1" ]]; then
+  nodeid="$1"
+fi
+
+ARCH=$(uname -p)
+if [[ "${ARCH}" == "x86_64" || ${ARCH} == "unknown" ]]; then
+  ARCH="amd64"
+fi
+
+if [ -z "$GITHUB_ACTIONS" ]
+then
+  # if not running on github actions, do not use sudo
```

Comment:
2nd time this is done, should use a helper function to execute with/out `sudo`. if all commands require `sudo`, have the calling script/workflow use `sudo` instead of bothering with it here